src/collection-preferences/content-display/utils.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { CollectionPreferencesProps } from '../interfaces';
+
+export function getSortedOptions({
+  options,
+  order,
+}: {
+  options: ReadonlyArray<CollectionPreferencesProps.ContentDisplayOption>;
+  order: ReadonlyArray<CollectionPreferencesProps.ContentDisplayItem>;
+}) {
+  const optionsSet: Record<string, CollectionPreferencesProps.ContentDisplayOption> = options.reduce(
```

Comment:
I think `optionsSet` is actually a map and `order` is not just order but also visibility. I'd change the code to make these observations better explicit:

```
function getSortedOptions({ options, optionsDisplay }: { ... }): readonly CollectionPreferencesProps.ContentDisplayOption[] {
  const getOption = new Map(options.map(option => [option.id, option]))
  const displayedOptions: CollectionPreferencesProps.ContentDisplayOption[] = []
  for (const { id } of optionsDisplay) {
    displayedOptions.push(getOption(id))
  }
  return displayedOptions;
}
```

Reply:
Made a refactor based on this in [ee0de7c](https://github.com/cloudscape-design/components/pull/804/commits/ee0de7c2e7d8ff3f97266ffeaefe46f69b9807e3), which also prevents walking the entire array again to check the visibility of each item.